### PR TITLE
retrieval pipeline fixes for new inference script

### PR DESCRIPTION
**Pull request by aniketrege**

### CODE REVIEW

File: inference/pytorch_inference.py [hunk 1]
```diff
@@ -21,18 +21,19 @@
 IMG_SIZE = 256
 CENTER_CROP_SIZE = 224
 NESTING_LIST=[2**i for i in range(3, 12)]
+NESTING_LIST=[8+(204*(i-1)) for i in range(1, 12)]
+print(NESTING_LIST)
```
Comment (adityakusupati) [inline inference/pytorch_inference.py]: Add a print comment that makes it clear.

File: retrieval/README.md [hunk 2]
```diff
@@ -6,35 +6,35 @@ We utilize a native [PyTorch inference](../inference/pytorch_eval.py) to generat
...
-for downstream adaptive retrieval or metric computation as, for example, `neighbors/exactl2_16dim-2048-NN_imagenet1k.csv`.
+for downstream adaptive retrieval or metric computation as, for example, `neighbors/mrl/exactl2_16dim_2048shortlist_1K.csv`.
 
 ## Adaptive Retrieval
-In an attempt to achieve equivalent performance to shortlisting at higher representation sizes with reduced MFLOPs, we perform adaptive retrieval by, for example, retrieving a shortlist $k = 200$ with rep. size $D_r = 16$ followed by reranking with a higher capacity representation $D_s = 2048$. The reranked k-NN shortlist is saved to disk as `8dim-reranked2048_200shortlist_imagenetv2_exactl2.csv` as shown in `reranking.ipynb`.
+In an attempt to achieve equivalent performance to shortlisting at higher representation sizes with reduced MFLOPs, we perform adaptive retrieval by, for example, retrieving a shortlist $k = 200$ with rep. size $D_r = 16$ followed by reranking with a higher capacity representation $D_s = 2048$ on ImageNetv2 representations. The reranked k-NN shortlist is saved to disk as `8dim-reranked16_200shortlist_V2_exactl2.csv` as shown in `reranking.ipynb`.
```
Comment (adityakusupati) [inline retrieval/README.md]: *ImageNetV2

File: utils.py [hunk 3]
```diff
@@ -12,11 +12,11 @@
 
 
 def get_ckpt(path):
-	ckpt=path 
+	ckpt=path
 	ckpt = torch.load(ckpt, map_location='cpu')
 	plain_ckpt={}
 	for k in ckpt.keys():
-		plain_ckpt[k[7:]] = ckpt[k] # torch DDP models have a extra wrapper of module., so to remove that... 
+		plain_ckpt[k[7:]] = ckpt[k] # remove the 'module' portion of key if model is Pytorch DDP
```
Comment (adityakusupati) [inline utils.py:19]: You can use soft load I think. Check "soft" in torch.load.

File: inference/pytorch_inference.py (hunk 1 above)
Comment (aniketrege) [reply inference/pytorch_inference.py]: This was leftover debugging code, removed in next patch

File: retrieval/README.md (hunk 2 above)
Comment (aniketrege) [reply retrieval/README.md]: updated

File: utils.py (hunk 3 above)
Comment (aniketrege) [reply utils.py:19]: I wasn't able to find this in torch documentation for `load` or `load_state_dict`. Should this current implementation be made more clear?

Comment (adityakusupati) [reply utils.py:19]: I am unable to find, but it should be OK.

**Review by adityakusupati — APPROVED** [review]

**Pull request merged**